### PR TITLE
feat(releases): Add release plan form

### PR DESCRIPTION
**Pull request by rottencandy**
## Fixes 
https://issues.redhat.com/browse/HAC-4700

## Description
Add release plan form.

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## How to test or reproduce?
Create release plan through release plan form.

## Browser conformance: 

- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/components/ReleaseService/ReleasePlan/ReleasePlanForm/RunReleasePipelineSection.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        }
+        required
+      />
+      {pipelineLocation === ReleasePipelineLocation.current && (
+        <>
+          <GitOptions />
+          <InputField
+            name="serviceAccount"
+            label="Service account"
+            helpText="Which service account do you want to use when constructing registry credentials?"
```
Comment (scoheb) [inline src/components/ReleaseService/ReleasePlan/ReleasePlanForm/RunReleasePipelineSection.tsx:79]:
```suggestion
            helpText="Which service account do you want to use to run your release pipeline?"
```

File: src/components/ReleaseService/ReleasePlan/ReleasePlanForm/RunReleasePipelineSection.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            labelIcon={
+              <HelpPopover
+                headerContent="Service account"
+                bodyContent="Which service account do you want to use when constructing registry credentials?"
```
Comment (scoheb) [inline src/components/ReleaseService/ReleasePlan/ReleasePlanForm/RunReleasePipelineSection.tsx:83]:
```suggestion
                bodyContent="Which service account do you want to use to run your release pipeline?"
```

File: src/components/ReleaseService/ReleasePlan/ReleasePlanForm/ReleasePlanForm.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+              />
+            }
+          />
+          <SwitchField
+            name="standingAttribution"
+            formLabel="Standing attribution"
+            labelIcon={
+              <HelpPopover
+                headerContent="Standing attribution"
+                bodyContent="The author of the release plan will be displayed to the release plan admission."
```
Comment (scoheb) [inline src/components/ReleaseService/ReleasePlan/ReleasePlanForm/ReleasePlanForm.tsx:87]:
```suggestion
                bodyContent="The author of the release plan will be provided as the standing attribution for releases."
```

File: src/types/coreBuildService.ts [hunk 4]
```diff
@@ -95,12 +95,22 @@ export type ReleasePlanKind = K8sResourceCommon & {
...
-  displayName: string;
   application: string;
   target: string;
+  data?: any;
+  pipelineRef?: {
+    resolver: ResolverType;
+    params: Param[];
+  };
+  serviceAccount?: string;
   releaseStrategy?: string;
```
Comment (johnbieren) [inline src/types/coreBuildService.ts:106]: releaseStrategy no longer exists

Comment (rottencandy) [reply src/types/coreBuildService.ts]: Thanks, removed. We will also have a separate ticket to remove all references to release strategy from the UI.

**Review by scoheb — APPROVED** [review]

**Comment by rottencandy** [comment]
Rebased to resolve conflicts.

**Comment by rottencandy** [comment]
@rohitkrai03 fixed.

File: src/components/ReleaseService/ReleasePlan/ReleasePlanForm/ReleasePlanFormPage.tsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        ? ReleasePipelineLocation.current
+        : ReleasePipelineLocation.target
+      : undefined,
+    serviceAccount: releasePlan?.spec?.serviceAccount ?? '',
+    target: releasePlan?.spec?.target ?? '',
+    data: releasePlan?.spec?.data ?? '',
+    params: releasePlanFormParams(releasePlan),
+    labels: releasePlan?.metadata?.labels
+      ? Object.entries(releasePlan?.metadata?.labels).map(([key, value]) => ({ key, value }))
+      : [{ key: '', value: '' }],
```
Comment (sahil143) [inline src/components/ReleaseService/ReleasePlan/ReleasePlanForm/ReleasePlanFormPage.tsx:109]: Please remove this empty key, value label. This is causing dry run error
image

Comment (sahil143) [reply src/components/ReleaseService/ReleasePlan/ReleasePlanForm/ReleasePlanFormPage.tsx:109]: Just pulled latest again it is fixed

**Comment by sahil143** [comment]
/lgtm

verified locally

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Pull request merged**